base/src/commonMain/kotlin/utils/BufferedAsListChannel.kt

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright 2023 Splendo Consulting B.V. The Netherlands
+ *
+ *     Licensed under the Apache License, Version 2.0 (the "License");
+ *     you may not use this file except in compliance with the License.
+ *     You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *     Unless required by applicable law or agreed to in writing, software
+ *     distributed under the License is distributed on an "AS IS" BASIS,
+ *     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *     See the License for the specific language governing permissions and
+ *     limitations under the License.
+ */
+
+package com.splendo.kaluga.base.utils
+
+import com.splendo.kaluga.base.singleThreadDispatcher
+import kotlinx.coroutines.CancellationException
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.channels.Channel
+import kotlinx.coroutines.channels.ChannelIterator
+import kotlinx.coroutines.channels.ChannelResult
+import kotlinx.coroutines.channels.ReceiveChannel
+import kotlinx.coroutines.channels.SendChannel
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.selects.SelectClause1
+import kotlinx.coroutines.selects.SelectClause2
+import kotlin.coroutines.CoroutineContext
+
+/**
```

Comment:
This is a pretty cool class, but I'm not sure it differs so _that_  much from looping `tryReceive` on the Channel after a `receive`. This could be thrown into an extension method `receiveBatch` or something.

I guess the main advantage is direct `asFlow` support?

This advantage should be weighed against how robust we think this class is in correctness and performance.